src/assets/scripts/aircraft/AircraftInstanceModel.js

```diff
@@ -1705,6 +1705,42 @@ export default class Aircraft {
     }
 
     /**
+     * Checks if the aircraft is inside the airspace of a specified airport
+     *
+     * @for AircraftInstanceModel
+     * @method isInsideAirspace
+     * @param  {airport} airport the airport whose airspace we are checking
+     * @return {Boolean}
+     * @private
+     */
+    isInsideAirspace(airport) {
+        let withinAirspaceLateralBoundaries = this.distance <= airport.ctr_radius;
+        const withinAirspaceAltitudeRange = this.altitude <= airport.ctr_ceiling;
+
+        if (!_isNil(airport.perimeter)) {    // polygonal airspace boundary
+            withinAirspaceLateralBoundaries = point_in_area(this.position, airport.perimeter);
+        }
+
+        return withinAirspaceAltitudeRange && withinAirspaceLateralBoundaries;
+    }
+
+    /**
+     * Aircraft has "weight-on-wheels" (on the ground)
+     * @for AircraftInstanceModel
+     * @method isOnGround
+     */
+    isOnGround() {
+        const error_allowance_ft = 5;
+        const apt = window.airportController.airport_get();
+        const rwy_elev = apt.getRunway(this.rwy_dep || this.rwy_arr).elevation;
```

Comment:
@n8rzz Yeah this whole block looks new but really it's all old; I just renamed it and then moved it so it was aligned alphabetically... but I've fixed the items you've pointed out with it.

And no, the `Runway` couldn't do it internally... the `Aircraft` maybe could, but in this particular case, I don't know if it would make much more sense. I could pull the `this.rwy_dep || this.rwy_arr` part into a variable, but other than that, I don't really know what else to do with it.